server/state-selectors-generator/index.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * External dependencies
+ */
+const path = require( 'path' );
+const fs = require( 'fs' );
+const chokidar = require( 'chokidar' );
+const chalk = require( 'chalk' );
+
+/**
+ * Internal dependencies
+ */
+const data = require( './data' );
+
+/*
+ * Constants
+ */
+const selectorsGlobalPath = path.normalize( 'client/state/selectors/' );
+const indexFileJs = path.join( selectorsGlobalPath, 'index.js' );
+const blacklist = /node_modules|\.git|index\.js|README.md|test/;
+const encoding = { encoding: 'utf-8' };
+const importBasePath = 'state/selectors';
+const logPrefix = chalk.yellow( 'selectors-generator: ' );
+
+/*
+ * Watcher instance
+ */
+let watcher;
+
+exports.init = function() {
+	// One-liner for current directory, ignores .dotfiles
+	watcher = chokidar.watch( selectorsGlobalPath, {
+		ignored: blacklist,
+		ignoreInitial: true,
+		depth: 0
+	} );
+
+	return this;
+};
+
+function addFileSelector( file ) {
+	console.log( logPrefix + `adding '${ file }' file selector` );
+	generateIndexFile( file );
+}
+
+function removeFileSelector( file ) {
+	console.log( logPrefix + `removing '${ file }' file selector` );
+	generateIndexFile( file );
+}
+
+/*
+ * Generate the content of the `index.js` file
+ * according to the selectors files.
+ */
+function generateIndexFile() {
+	let indexContent = '';
+
+	fs.readdir( selectorsGlobalPath, ( err, files ) => {
+		const total = files.length - 1;
+		let counter = 0;
+
+		files.forEach( file => {
+			counter++;
+			if ( ! file.match( blacklist ) ) {
+				file = file.replace( /\.jsx?/g, '' );
+
+				const moduleName = file.replace( /-([a-z])/g, function( str ) {
+					return str[ 1 ].toUpperCase();
+				} );
+
+				const exportLine = `export ${ moduleName } from '${ importBasePath }/${ file }';`;
+				indexContent += exportLine + '\n';
+
+				if ( counter === total ) {
+					saveIndexFile( indexContent );
+				}
+			}
+		} );
+	} );
+}
```

Comment:
just as a note, I think we could benefit by further breaking apart these functions into separate parts of this file. you've done a nice job modularizing the code! let's take that one step further and create a separate function for handling each individual file